### PR TITLE
issue: 4007026 - [CI] Release pipeline

### PR DESCRIPTION
**Pull request by NirWolfer**
## Description
Unti now, release process is done manually by the dev team. This process involves creating a tag on a commit that needs to be released, building that tag and copying the files to NFS location

##### What
The idea is to create a jenkins pipeline that will do the entire process with given inputs, This commit will commence phase 1 - doing the build and copying of packages created to NFS.
THE CREATION OF A RELEASE TAG ON A COMMIT IS NOT PART OF THIS PR

##### Why ?
[HPCINFRA-1889](https://jirasw.nvidia.com/browse/HPCINFRA-1889)

##### How ?
Add release_jjb and the corosponding matrix file for a new pipeline definition, the release is run on rhel8.6
Add do_release.sh script to run the build and NFS copy to release the package
Add Dockerfile for release container with all dependencies installed

## Change type
What kind of change does this PR introduce?
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] CI related changes
- [ ] Documentation content changes
- [ ] Tests
- [ ] Other

## Check list
- [ ] Code follows the style de facto guidelines of this project
- [ ] Comments have been inserted in hard to understand places
- [ ] Documentation has been updated (if necessary)
- [ ] Test has been added (if possible)

### CODE REVIEW

File: .ci/pipeline/release_jjb.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+            name: "release_tag"
+            default: ""
+            description: "Tag to release"
+        - string:
+            name: sha1
+            default: 'tags/$release_tag'
+            description: 'commit to use, defaults to the commit tag provided under release_tag'
+        - string:
+            name: "revision"
+            default: "0"
```
Comment (dpressle) [inline .ci/pipeline/release_jjb.yaml:28]: should be 1 not 0

File: .ci/do_release.sh [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+else
+     echo "do_release is set to false, skipping package release."
+fi
+
+set +x
+
+echo "**********************************"
+echo "Finished do_release.sh script..."
+if [ "${do_release}" = true ] ; then
```
Comment (dpressle) [inline .ci/do_release.sh:86]: why not put this in the do_release block it feels like duplicating blocks

**Review by dpressle — CHANGES_REQUESTED** [review]

**Comment by NirWolfer** [comment]
bot:retest

File: .ci/pipeline/release_jjb.yaml (hunk 1 above)
Comment (NirWolfer) [reply .ci/pipeline/release_jjb.yaml:28]: This is whats currently under the vma release folder (all revision numbers are set to 0)
We need to check if they want it changed to starting with 1 instead

**Comment by AlexanderGrissik** [comment]
dpressle please approve ..

File: .ci/pipeline/release_jjb.yaml (hunk 1 above)
Comment (dpressle) [reply .ci/pipeline/release_jjb.yaml:28]: So fix the default revision in the script, they must match

File: .ci/do_release.sh [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    echo "ERROR: FULL_VERSION: ${configure_ac_version} from configure.ac doesn't match tag: ${release_tag} provided! Exit"
+    exit 1
+fi
+
+if [ "${do_release}" = true ] ; then
+    echo "do_release is set to true, will release package into ${release_folder}/libvma_v_${release_tag}-${revision}"
+
+    cd pkg/packages || { echo "pkg folder is missing, exiting..."; exit 1; }
+    pkg_name=$(ls -1 libvma-"${release_tag}"-"${revision}".src.rpm)
+    DST_DIR=${release_folder}/libvma_v_${release_tag}-${revision}
```
Comment (dpressle) [inline .ci/do_release.sh:64]: looking at the release folder and this is not the path / names convention

Existing one today:
```
$ ls -ltr /auto/mswg/release/vma/vma_v_9.8.60-0/src/
-rw-r--r-- 1 danielli labop 1600185 Feb 28 16:08 libvma-9.8.60-1.src.rpm
```

and yours is (look at the folder name and the missing src folder
```
$ ls -ltr /auto/mswg/release/vma/tmp/libvma_v_9.8.60-0
-rw-r--r-- 1 swx-jenkins labop 1602856 Jul 31 15:25 libvma-9.8.60-0.src.rpm
```

**Review by dpressle — CHANGES_REQUESTED** [review]

**Review by dpressle — APPROVED** [review]

**Pull request merged**